src/dune_rules/modules.ml

```diff
@@ -527,29 +510,24 @@ let rec find_dep t ~of_ name =
     | Impl_or_lib -> Some m
     | Vlib -> Option.some_if (Module.visibility m = Public) m
 
-let singleton_exe m =
+let make_singleton m mangle =
   Singleton
-    (let mangle = Mangle.Exe in
-     let main_module_name = (Mangle.prefix mangle).public in
+    (let main_module_name = (Mangle.prefix mangle).public in
      Module.with_wrapper m ~main_module_name)
 
-let exe_unwrapped m = Unwrapped m
-
-let exe_wrapped ~src_dir ~modules =
-  match as_singleton modules with
-  | Some m -> singleton_exe m
-  | None -> Wrapped (Wrapped.exe ~src_dir ~modules)
+let singleton_exe m = make_singleton m Exe
 
-let singleton_melange m =
-  Singleton
-    (let mangle = Mangle.Melange in
-     let main_module_name = (Mangle.prefix mangle).public in
-     Module.with_wrapper m ~main_module_name)
+let exe_unwrapped m = Unwrapped m
 
-let melange_wrapped ~src_dir ~modules =
+let make_wrapped ~src_dir ~modules kind =
+  let mangle : Mangle.t =
+    match kind with
+    | `Exe -> Exe
+    | `Melange -> Melange
+  in
   match as_singleton modules with
-  | Some m -> singleton_melange m
-  | None -> Wrapped (Wrapped.melange ~src_dir ~modules)
+  | Some m -> make_singleton m mangle
+  | None -> Wrapped (Wrapped.gen_exe ~src_dir ~modules mangle)
```

Comment:
I am thinking of a name to replace "exe" that covers both melange emit entry points and executables... "program"? "application"?

```suggestion
  | None -> Wrapped (Wrapped.gen_program ~src_dir ~modules mangle)
```

Reply:
Can't say I'm a fan of the overly generic "program" either. I'll just stick with `exe_or_melage` at least it's specific even if verbose.